### DISCUSSION TITLE
traceIdRatioBased Logic

### DISCUSSION THREAD

**Discussion by alokmajumder** [open-telemetry/opentelemetry-java#2877] [Q&A]
Hi traceIdRatioBased how it works? When I put 0.5 I observed per-second wise 50% traced. Is my assumption correct. Or it is on minute?

**Comment by anuraaga** [comment]
Hi @alokmajumder - traceIdRatio is currently a purely random sampling at each instant. 0.5 means a 50% probability in the long term - but as with any coin flip there would still be a chance of 5 heads in a row. More importantly than the time, e.g., second vs minutes, is the number of requests. More requests will mean a distribution closer to the set probability.
